app/src/main/java/com/xa/xpensauditor/Day_wise_visualization.java

```diff
@@ -0,0 +1,204 @@
+        List<Integer> arr=new ArrayList<>();
+
+        Cartesian cartesian = AnyChart.column();
+
+        List<DataEntry> data = new ArrayList<>();
+        {
+
+            Boolean f = false;
+            Calendar calendar = Calendar.getInstance();
+
+            int thisYear = calendar.get(Calendar.YEAR);
+            // Log.d(TAG, "# thisYear : " + thisYear);
+
+            int thisMonth = calendar.get(Calendar.MONTH)+1;
+            //Log.d(TAG, "@ thisMonth : " + thisMonth);
+
+            int thisDay = calendar.get(Calendar.DAY_OF_MONTH);
+            //Log.d(TAG, "$ thisDay : " + thisDay);
+
+
+            if (thisYear > y2) {
+                f = true;
+            } else {
+                if (thisYear == y2) {
+                    if (thisMonth > m2) {
+                        f = true;
+                    } else {
+                        if (thisMonth == m2) {
+                            if (thisDay >= d2) {
+                                f = true;
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (f) {
+System.out.println("Day_Wise   "+m2 +"    "+y2);
+                DatabaseReference reference= FirebaseDatabase.getInstance().getReference().child(Uid).child("DateRange").child(String.valueOf(m2 + "-" + y2)).child("Transactions");
+                reference.addValueEventListener(new ValueEventListener() {
+                    List<DataEntry> data = new ArrayList<>();
+                    String dayy,dumm;
+                    int dayy1,dumm1;
+                    int dayyy;
+
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot datasnapshot) {
+
+                        for(DataSnapshot snapshot :datasnapshot.getChildren() ){
+                            String amm=snapshot.child("Amount").getValue().toString();
+                            System.out.println(" Finallyyy   "+amm);
+                            int am1=Integer.valueOf(amm);
+                            tot=am1+tot;
+      dumm=snapshot.child("Day").getValue().toString();
+      dumm1=Integer.valueOf(dumm);
+      System.out.println("dummmmm   "+dumm1+"    "+dayyy);
+      if(dayyy==dumm1){
+          System.out.println("inside");
+          int temp=0;
+          temp=hm.get(dayyy);
+      temp=temp+am1;
+          System.out.println("dummmmm1   "+dayyy+"    "+temp);
+      hm.put(dayyy,temp);
+      }
+      else{
+      dayy=snapshot.child("Day").getValue().toString();
+      dayyy=Integer.parseInt(dayy);
+     hm.put(dayyy,am1);}
+                 }
+
+                        ArrayList<Integer> sortedKeys
+                                = new ArrayList<Integer>(hm.keySet());
+
+                        Collections.sort(sortedKeys);
+
+                        // Display the TreeMap which is naturally sorted
+                        for (Integer x : sortedKeys) {
+                            data.add(new ValueDataEntry(x,hm.get(x)));
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for' child has incorrect indentation level 28, expected level should be one of the following: 22, 24, 26.